src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,16 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        User storagedUser = storageDao.get(user.getLogin());
+        if (user == null
+                || user.getPassword() == null
+                || user.getLogin() == null
+                || (storagedUser != null && user.getLogin().equals(storagedUser.getLogin()))
+                || user.getAge() == null
+                || user.getAge() < 18) {
+            throw new RuntimeException();
+        }
```

Comment:
split this big into several small ones + add informative messages inside of the RuntimeException's constructor